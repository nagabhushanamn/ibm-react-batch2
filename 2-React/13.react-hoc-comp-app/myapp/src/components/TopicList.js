
import React, { Component } from 'react';
import DataSource from '../ds/DataSource'

import { withSubscription } from '../hoc/index';

class TopicList extends Component {


    // constructor(props) {
    //     super(props);
    //     this.state = {
    //         //"DataSource" is some global data source
    //         topics: DataSource.getTopics()
    //     };
    // }

    // componentDidMount() {
    //     // Subscribe to changes
    //     DataSource.addChangeListener(this.handleChange);
    // }

    // componentWillUnmount() {
    //     // Clean up listener
    //     DataSource.removeChangeListener(this.handleChange);
    // }

    // handleChange() {
    //     // Update component state whenever the data source changes
    //     this.setState({
    //         comments: DataSource.getTopics()
    //     });
    // }

    render() {
        let { data } = this.props;
        let { onTopicSelect } = this.props;
        let topicElements = data.map((topic, idx) => {
            return (
                <div onClick={() => { onTopicSelect(topic) }}
                    key={idx}
                    className="list-group-item">
                    {topic}
                </div>);
        });
        return (
            <div className="panel panel-default">
                <div className="panel-heading">All Topics</div>
                <div className="panel-body">
                    <div className="list-group col-md-6 col-sm-6">
                        {topicElements}
                    </div>
                </div>
            </div>

        );
    }
}

// export default TopicList;


export default withSubscription(TopicList, (DataSource, props) => { return DataSource.getTopics() });