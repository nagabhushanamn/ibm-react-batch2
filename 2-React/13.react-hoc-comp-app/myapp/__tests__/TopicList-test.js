import TopicList from '../src/components/TopicList';
import React from 'react';
// import { mount, shallow } from 'enzyme';


// test('TopicList component renders the panel with title All Topics , shud return Topic onClick', () => {

//   const handleSelectTopic = jest.fn(); // mock

//   const wrapper = mount(
//     <TopicList onTopicSelect={handleSelectTopic} data={["React", "Redux"]} />
//   );

//   const titleDiv = wrapper.find('.panel-heading');
//   expect(titleDiv.text()).toBe('All Topics');

//   const topicEle = wrapper.find('.list-group-item').first();
//   topicEle.simulate('click');
//   expect(handleSelectTopic).toBeCalledWith('React')

// });




// // snapshot Testing  ( unique )


import renderer from 'react-test-renderer';

describe('TopicList component renders the topics correctly', () => {
  
  it('renders correctly', () => {

   // snapshop 
    const rendered = renderer.create(
      <TopicList  />
    );

    expect(rendered.toJSON()).toMatchSnapshot();

  });
});