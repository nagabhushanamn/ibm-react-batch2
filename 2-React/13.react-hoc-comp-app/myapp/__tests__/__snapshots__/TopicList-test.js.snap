// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicList component renders the topics correctly renders correctly 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    All Topics
  </div>
  <div
    className="panel-body"
  >
    <div
      className="list-group col-md-6 col-sm-6"
    >
      <div
        className="list-group-item"
        onClick={[Function]}
      >
        React
      </div>
      <div
        className="list-group-item"
        onClick={[Function]}
      >
        Redux
      </div>
      <div
        className="list-group-item"
        onClick={[Function]}
      >
        Jest
      </div>
      <div
        className="list-group-item"
        onClick={[Function]}
      >
        JavaScript
      </div>
      <div
        className="list-group-item"
        onClick={[Function]}
      >
        Java
      </div>
    </div>
  </div>
</div>
`;
